lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -673,6 +674,23 @@ newDBLayer trace defaultFieldValues mDatabaseFile = do
         , readProtocolParameters =
             \(PrimaryKey wid) -> selectProtocolParameters wid
 
+        {-----------------------------------------------------------------------
+                                 Delegation Rewards
+        -----------------------------------------------------------------------}
+
+        , putDelegationRewardBalance =
+            \(PrimaryKey wid) (Quantity amt) -> ExceptT $ do
+            selectWallet wid >>= \case
+                Nothing -> pure $ Left $ ErrNoSuchWallet wid
+                Just _  -> Right <$> repsert
+                    (DelegationRewardKey wid)
+                    (DelegationReward wid amt)
```

Comment:
Yes - I think "eventually correct" is the best we that we can do, because it's not feasible to synchronise wallet applyBlock updates with the LocalStateQuery result messages.